overlord/devicestate/devicestate.go

```diff
@@ -461,10 +504,9 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, ms modelSna
 			changed = ms.currentModelSnap.PinnedTrack != ms.newModelSnap.PinnedTrack
 		}
 		if changed {
-			// new modes specifies the same snap, but with a new channel
```

Comment:
why removing the comment? also what should happen in this case for a local remodel? in theory we could let people not spefify a snap which means keeping the current one but just switching the channel?

Reply:
same question for non-essential snaps

Reply:
ok, we'll need to document this because there also cases were keeping the same snap can happen, for example switching from a latest track to a versioned track for some app